contracts/Comet.sol

```diff
@@ -764,47 +784,27 @@ contract Comet is CometMainInterface {
      * @dev Note: Safely handles non-standard ERC-20 tokens that do not return a value. See here: https://medium.com/coinmonks/missing-return-value-bug-at-least-130-tokens-affected-d67bf08521ca
      */
     function doTransferIn(address asset, address from, uint amount) internal returns (uint) {
-        uint256 preTransferBalance = IERC20NonStandard(asset).balanceOf(address(this));
-        IERC20NonStandard(asset).transferFrom(from, address(this), amount);
-        bool success;
-        assembly {
-            switch returndatasize()
-                case 0 {                       // This is a non-standard ERC-20
-                    success := not(0)          // set success to true
-                }
-                case 32 {                      // This is a compliant ERC-20
-                    returndatacopy(0, 0, 32)
-                    success := mload(0)        // Set `success = returndata` of override external call
-                }
-                default {                      // This is an excessively non-compliant ERC-20, revert.
-                    revert(0, 0)
-                }
+        _nonReentrantBefore();
+        uint256 preTransferBalance = ERC20(asset).balanceOf(address(this));
+        (bool success, bytes memory returndata) = asset.call(abi.encodeWithSelector(ERC20.transferFrom.selector, from, address(this), amount));
+        if (!success || !(returndata.length == 0 || abi.decode(returndata, (bool)))) {
+            revert TransferInFailed();
         }
-        if (!success) revert TransferInFailed();
-        return IERC20NonStandard(asset).balanceOf(address(this)) - preTransferBalance;
+        _nonReentrantAfter();
+        return ERC20(asset).balanceOf(address(this)) - preTransferBalance;
     }
 
     /**
      * @dev Safe ERC20 transfer out
      * @dev Note: Safely handles non-standard ERC-20 tokens that do not return a value. See here: https://medium.com/coinmonks/missing-return-value-bug-at-least-130-tokens-affected-d67bf08521ca
      */
     function doTransferOut(address asset, address to, uint amount) internal {
-        IERC20NonStandard(asset).transfer(to, amount);
-        bool success;
-        assembly {
-            switch returndatasize()
-                case 0 {                       // This is a non-standard ERC-20
-                    success := not(0)          // set success to true
-                }
-                case 32 {                      // This is a compliant ERC-20
-                    returndatacopy(0, 0, 32)
-                    success := mload(0)        // Set `success = returndata` of override external call
-                }
-                default {                      // This is an excessively non-compliant ERC-20, revert.
-                    revert(0, 0)
-                }
+        _nonReentrantBefore();
```

Comment:
I'm leaning on having the reentrancy guards be on the `external` functions like `supply` and `withdraw` rather than the internal `doTransferIn/Out` functions. It's easier to reason about which functions need the guard, and we also avoid having to set up the guards twice in functions that transfer multiple times like `buyCollateral`.

Reply:
I tried to add to  
`supplyInternal` and `withdrawInternal` and `transferInternal` and `buyCollateral` 

But unfortunately :( the compiled bytecode size will be somewhat 24.8  :(

Reply:
With the latest changes on Custom error, and nonReentrant on all exeternal function the contract size became: 
```
|  CometFactory                    ·      24.745  ·        +0.200  │
 ···································|··············|·················
 |  CometModifiedFactory            ·      24.883  ·        +0.200  │

```

Both are above 24.576Kb :(